### PR TITLE
Remove npc_publisher from scenario.launch.xml

### PR DESCRIPTION
**Pull request by nnmm**
It's now an inline node.

### CODE REVIEW

File: api/scenario_api/launch/scenario.launch.xml [hunk 1]
```diff
@@ -41,11 +41,6 @@
   </group>
 
   <!-- Dummy Perception -->
-  <group>
-    <include file="$(find-pkg-share npc_simulator)/launch/npc_simulator.launch.xml">
-      <arg name="initial_engage_state" value="$(var initial_engage_state)" />
```
Comment (fred-apex-ai) [inline api/scenario_api/launch/scenario.launch.xml:46]: looks ok. Do you still pass the `initial_engage_state` to the npc_simulator?

Comment (nnmm) [reply api/scenario_api/launch/scenario.launch.xml:46]: We should pass it to `scenario_api`, but that's not part of this launch file. In fact it seems to be part of no real launch file, only the sample launch file next door. I really need someone to explain this whole design to me.

Comment (nnmm) [reply api/scenario_api/launch/scenario.launch.xml:46]: Ah, wait, it's part of the `scenario_runner` executable. `scenario_api` has an executable too but it's never used apart from the sample launch file.

Comment (nnmm) [reply api/scenario_api/launch/scenario.launch.xml:46]: I'd add it there later. I'll need to do work with `scenario_runner` anyway I think.

Comment (fred-apex-ai) [reply api/scenario_api/launch/scenario.launch.xml:46]: @nnmm I can hear your frustration about this complex 'who calls whom'. I agree it should be disentangled before it's ready to into autoware.auto

**Pull request merged**